src/gc-stock.c

```diff
@@ -3595,6 +3595,28 @@ JL_DLLEXPORT uint64_t jl_gc_get_max_memory(void)
 
 // allocation wrappers that track allocation and let collection run
 
+JL_DLLEXPORT void jl_gc_heap_increment(size_t sz)
+{
+    jl_gcframe_t **pgcstack = jl_get_pgcstack();
+    jl_task_t *ct = jl_current_task;
+    if (pgcstack != NULL && ct->world_age) {
```

Comment:
Why does world_age matter here? or is that just a proxy to see if it's a valid task?

Reply:
Yeah this code looked weird to me as well, but I assume it's a valid task check, maybe an artifact before thread-adoption?